### ISSUE TITLE
Wrong item order in ACCEPT_KEYWORDS

### ISSUE THREAD

**Issue by xofyarg** [vaeth/eix#76]
I'm using gentoo prefix, the default ACCEPT_KEYWORDS from profile is `~${ARCH}`, my local config in make.conf has `ACCEPT_KEYWORDS="-~${ARCH} ${ARCH}"`, this is to make the system default to `"${ARCH}"`. emerge can understand it without any problem. But eix still treat the system as `~${ARCH}`.

```
$ emerge --info | grep ACCEPT_KEYWORDS
ACCEPT_KEYWORDS="amd64"

$ eix --print LOCAL_PORTAGE_CONFIG
true

$ eix --print ACCEPT_KEYWORDS
-~amd64 amd64 amd64 ~amd64
```

**Comment by vaeth** [comment]
Thanks for the report. Is the issue fixed in master?
(You can install 0.34.5_alpha200906 or 99999999 from the mv repository for testing.)

**Comment by xofyarg** [comment]
`0.34.5_alpha200906` fixed the issue, thank you!

**Comment by vaeth** [comment]
Thanks for testing.

**Comment by xofyarg** [comment]
There's a hiccup though:
```
$ eix-sync
 * push.sh from https://github.com/vaeth/push/ is needed
```

Created a PR: https://github.com/vaeth/push/pull/2

**Comment by vaeth** [comment]
Reopening, since the original fix does not work:

People complained (rightfully) that after changing of the read order PORTDIR_OVERLAY is not honored to find profiles.
So there is a chicken-and-egg problem.

Possible solutions:

1. Remember and delete all incremental (accumulative) variables after reading make.conf and append them at the very end
   (this happened previously with USE, and ACCEPT_KEYWORDS is incremental, too).
2. Remember and delete all variables except the known "early" variables (currently PORTAGE_PROFILE, PORTAGE_REPOSITORIES, PORTDIR, PORTDIR_OVERLAY) and treat all the rest later.

In both cases there is the variant whether or not to handle make.globals separately.

The solution should probably mimick portage as far as possible. Not sure yet how to test what portage does...

**Comment by vaeth** [comment]
Experiments show for random (non-incremental variables):

(a) make.globals is overridden by profile which in turn is overridden by make.conf
(b) make.conf cannot use variables from make.globals

So it seems that the approach from 2 is correct: Sourcing all variables from make.conf into a fresh environment (not inherited from make.globals), and use this environment in the beginning for overriding the known early variables and in the end for overriding anything else (appending to incremental variables).

**Comment by vaeth** [comment]
A version with the above code is in master. Version 0.34.6_alpha20200907 of the mv overlay refers to it.
Testing is appreciated.

**Comment by xofyarg** [comment]
Looks a bit weird with `0.34.6_alpha20200907`, and it doesn't work as expected.

```
$ eix --print ACCEPT_KEYWORDS
amd64 ~amd64
-~
```

**Comment by vaeth** [comment]
That there is a newline instead of a space is to be expected, since I use now the same algorithm as for appending environment variables. Probably, I should use the same symbol always; probably always "\n".

What is causing the error is that the next line does not read `-~amd64`.
Obviously, you are using $ARCH in make.conf.

I checked, and portage indeed has the variables from profile available in make.conf. Bummer!

This means that there is no other solution for the chicken-and-egg problem except sourcing make.conf twice:
First, to get the early variables (for finding the profile paths), and then again **after** sourcing the profile to have the variables needed to source make.conf properly.

This is a considerable delay in starting eix, but I see no other solution...

**Comment by vaeth** [comment]
Reading make.conf twice is now implemented in the master branch (0.34.6_alpha20200907 in the mv overlay).

**Comment by xofyarg** [comment]
Appreciate for the fix, it works for me now.

**Comment by vaeth** [comment]
Thanks for testing again. Released as eix-0.34.6
